### PR TITLE
Database Handler - Redis

### LINKED ISSUE #4: Database handler: Redis
Backups for the Redis database.

### PR DESCRIPTION
**Pull request by ks129**
- Implemented Redis database handler that is using `redis-cli` with `--rdb` kwarg to store backups as `.rdb` file. This takes connstring in format `redis://password@host:port` (this doesn't need username).
- Added documentation about this also to README.md.
- Added Redis container to `docker-compose.yml` for local development.
- (Other): Removed `Blackbox.loggers` from `_get_all_connstrings` because this don't exist and give error.

Closes #4

### CODE REVIEW

**Review by lemonsaurus — CHANGES_REQUESTED** [review]
This looks very good, @ks129! I have a few requests, though.

### What happens if the redis server doesn't require a password?
In that case, the user might be tempted to use a connstring like `redis://@host:port`, but this will not match the regex, and so blackbox will break.

If the password is optional, then maybe we should make the user pass it in as an optional parameter. So, the connstring would then be something like `redis://host:port?password=pw`.

If you agree, make this change. Otherwise, explain why you don't agree.

### _Have you actually tested this?_
- There are no unit tests. That might be okay, but then it needs to be tested carefully by humans.
- There's no initial data for the redis, like we've done for mongo and postgres. This means it will be hard to verify that the backup actually has the data it should have when we test this manually.

Please consider setting up some initial data for Redis, and write unit tests if you want to (not required), and let me know how much manual testing you've done in your reply to this comment.

**Comment by ks129** [comment]
@lemonsaurus 
- Added and wrote documentation for the script that inserts testing data to Docker Redis.
- Moved Redis' password to extra parameters.
- Capture both stdout and stderr for command output.
- Use :x: for field value when a database backup fails. Discord doesn't accept empty values.

I'd like to not write unit tests. This isn't my strong side.

About my test cases:
1.
- Inserted data with the script.
- Filled all config properly.
- Ran script.
- Downloaded backup from S3-like storage.
- Replaced local Redis instance RDB with backup RDB.
- Checked is all values correct (yes, they were).
2.
- Inserted data with the script.
- Didn't provide password when password was required.
- Ran script.
- Got unauthenticated error in Discord output.
3.
- Inserted data with the script.
- Provided password when password wasn't required.
- Ran script.
- Backup succeed.

File: blackbox/handlers/notifiers/discord.py [hunk 1]
```diff
@@ -39,6 +39,10 @@ def _parse_report(self, report: dict) -> dict:
                 emoji = ":white_check_mark:" if provider['success'] else ":x:"
                 field['value'] += f"{emoji}  {provider['type']}\n"
 
+            # When backup failed, add X emoji to value to avoid error.
```
Comment (lemonsaurus) [inline blackbox/handlers/notifiers/discord.py:42]: Which error are we avoiding? is this really happening "when backup failed"? This comment could be improved.

File: docker-compose.yaml [hunk 2]
```diff
@@ -28,3 +28,9 @@ services:
     volumes:
       - ./docker/mongo/init_data:/init_data
       - ./docker/mongo/mongo_import.sh:/docker-entrypoint-initdb.d/mongo_import.sh
+
+  redis_blackbox:
```
Comment (lemonsaurus) [inline docker-compose.yaml:32]: why not just `redis`, following the convention from the rest of this file? We're giving it a different container_name anyway, so it shouldn't conflict with anything.

File: docker/redis/start-redis.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# Start Redis in background and wait 1 second
```
Comment (lemonsaurus) [inline docker/redis/start-redis.sh:3]: If we're borrowing this whole file from that Stack Overflow answer, we should give it some credit. Maybe a comment near the top of this file that links to the answer and mentions that we borrowed it from there?

**Review by lemonsaurus — CHANGES_REQUESTED** [review]
Starting to look pretty good. A couple of minor things.

Your manual tests seem fine.

**Review by lemonsaurus — APPROVED** [review]
One last language change, and we can merge.

**Pull request merged**